privacyidea/lib/eventhandler/tokenhandler.py

```diff
@@ -372,8 +374,15 @@ def do(self, action, options=None):
                                                 d.strftime(DATE_FORMAT))
                 elif action.lower() == ACTION_TYPE.SET_FAILCOUNTER:
                     try:
-                        set_failcounter(serial,
-                                        int(handler_options.get("fail counter")))
+                        handler_option = handler_options.get("fail counter")
+                        if handler_option.startswith("+"):
+                            token_failcount = get_one_token(serial=serial).token.failcount
+                            set_failcounter(serial, token_failcount + int(handler_option[1:]))
```

Comment:
You already have to token_object ``token_object``. So if you are calling ``set_cailcounter`` with the serial number, the code will do a second token lookup.
You can save the 2nd token lookup if you do:

~~~~python
token_object.set_failcount(token_oject.token.failcount + int(handler_option[1:]))
~~~~

...
or we could enhance ``lib/token.py:set_failcounter`` to also directly accept increments or decrements.

Reply:
ah, thank's